changeset:   7293:4fc233f0c1d9
tag:         bugs-gentoo/0007-ncurses-matching-tinfo.patch
tag:         qtip
tag:         tip
user:        Fabian Groffen <grobian@gentoo.org>
date:        Sat Mar 31 14:04:51 2018 +0200
files:       configure.ac
description:
configure: check for tinfo matching ncurses

When we selected ncursesw, look for tinfow as well.

Bug: https://bugs.gentoo.org/651552


diff -r ded90ccbc6de -r 4fc233f0c1d9 configure.ac
--- a/configure.ac	Sat Sep 10 11:48:50 2016 +0100
+++ b/configure.ac	Sat Mar 31 14:04:51 2018 +0200
@@ -299,12 +299,13 @@ main ()
         AC_CHECK_LIB($cf_ncurses, initscr,
                 [MUTTLIBS="$MUTTLIBS -l$cf_ncurses"
 
-                AC_CHECK_LIB(tinfo, tgetent, [MUTTLIBS="$MUTTLIBS -ltinfo"])
-
                 if test "$cf_ncurses" = ncursesw; then
+                  AC_CHECK_LIB(tinfow, tgetent, [MUTTLIBS="$MUTTLIBS -ltinfow"])
                   AC_CHECK_HEADERS(ncursesw/ncurses.h,[cf_cv_ncurses_header="ncursesw/ncurses.h"],
                     [AC_CHECK_HEADERS(ncurses.h,[cf_cv_ncurses_header="ncurses.h"])])
                 else
+                  AC_CHECK_LIB(tinfo, tgetent, [MUTTLIBS="$MUTTLIBS -ltinfo"])
+
                   AC_CHECK_HEADERS(ncurses/ncurses.h,[cf_cv_ncurses_header="ncurses/ncurses.h"],
                     [AC_CHECK_HEADERS(ncurses.h,[cf_cv_ncurses_header="ncurses.h"])])
 		fi],

